experiments/kafka-hbase-pipeline/src/main/scala/org/apache/gearpump/experiments/pipeline/PipeLineDSL.scala

```diff
@@ -0,0 +1,157 @@
+import org.apache.gearpump.streaming.transaction.api.MessageDecoder
+import org.apache.gearpump.util.{Constants, LogUtil}
+import org.apache.hadoop.conf.Configuration
+import org.slf4j.Logger
+import upickle._
+
+import scala.util.Try
+
+class ReplayableSource(kafkaConfig: KafkaConfig) extends Traverse[Array[Datum]] {
+  private val batchSize = kafkaConfig.getConsumerEmitBatchSize
+  private val msgDecoder: MessageDecoder = kafkaConfig.getMessageDecoder
+  lazy val source = new KafkaSource(kafkaConfig.getClientId, TaskId(0,0), 1, kafkaConfig, msgDecoder)
+  override def foreach[U](fun: Array[Datum] => U): Unit = {
+    val list = source.pull(batchSize)
+    list.foreach(msg => {
+      val jsonData = msg.msg.asInstanceOf[String]
+      val envelope = read[Envelope](jsonData)
+      val body = read[Body](envelope.body)
+      val metrics = body.metrics
+      fun(metrics)
+    })
+  }
+}
+
+object PipeLineDSL extends App with ArgumentsParser {
+  private val LOG: Logger = LogUtil.getLogger(getClass)
+  val PROCESSORS = "pipeline.processors"
+  val PERSISTORS = "pipeline.persistors"
+
+  override val options: Array[(String, CLIOption[Any])] = Array(
+    "conf" -> CLIOption[String]("<conf file>", required = true)
+  )
+  val context = ClientContext()
+  implicit val system = context.system
+
+  def toHBaseSink(taskContext: TaskContext, userConfig: UserConfig): Array[Datum] => Unit = {
+    val LOG: Logger = LogUtil.getLogger(taskContext.getClass)
+    val repo = new HBaseRepo {
+      def getHBase(table: String, conf: Configuration): HBaseSinkInterface = HBaseSink(table, conf)
+    }
+    val config: Config = userConfig.getValue[PipeLineConfig](PIPELINE).get
+    val hbaseConsumer = HBaseConsumer(taskContext.system, Some(config))
+    def hbaseSink(hbase: HBaseSinkInterface)(metrics: Array[Datum]): Unit = {
+      LOG.info("success")
+      metrics.foreach(datum => {
+        hbase.insert(System.currentTimeMillis.toString, hbaseConsumer.table, hbaseConsumer.table, datum.value.toString)
+      })
+    }
+    hbaseSink(hbaseConsumer.getHBase(repo))
+  }
+
+  def application(context: ClientContext, config: ParseResult): Unit = {
+    val pipeLinePath = config.getString("conf")
+    val pipeLineConfig = PipeLineConfig(ConfigFactory.parseFile(new java.io.File(pipeLinePath)))
+    val persistors = pipeLineConfig.getInt(PERSISTORS)
+    val kafkaConfig = KafkaConfig(pipeLineConfig)
+    val repo = new HBaseRepo {
+      def getHBase(table: String, conf: Configuration): HBaseSinkInterface = HBaseSink(table, conf)
+    }
+    val appConfig = UserConfig.empty.withValue(KafkaConfig.NAME, kafkaConfig).withValue(PIPELINE, pipeLineConfig).withValue(HBASESINK, repo)
+    System.setProperty(Constants.GEARPUMP_CUSTOM_CONFIG_FILE, pipeLinePath)
+
+    val app = StreamApp("PipeLineDSL", context, appConfig)
+    val producer = app.readFromKafka(new ReplayableSource(kafkaConfig), 1, "time-replayable-producer")
```

Comment:
Can this be simplified by only relying on a kakfaConfig?

For example:
app.readfromKafka(kafkaConfig, name)